src/v2/view-builder/views/idp/AuthenticatorIdPView.js

```diff
@@ -0,0 +1,25 @@
+import EnrollIdPForm from './EnrollAuthenticatorIdPForm';
+import ChallengeIdpForm from './ChallengeAuthenticatorIdPForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import AuthenticatorVerifyFooter from '../../components/AuthenticatorVerifyFooter';
+
+export default BaseAuthenticatorView.extend({
+  initialize () {
+    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
+
+    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
+    const existingAuthenticator = authenticatorEnrollments
+      .filter(authenticator => authenticator.id === currentAuthenticator.id);
+
+    // Set the view based on whether or not we're in a challenge flow
+    this.Body = existingAuthenticator.length
+      ? ChallengeIdpForm
+      : EnrollIdPForm;
+
+    this.Footer = existingAuthenticator.length
```

Comment:
AuthenticatorVerifyFooter and AuthenticatorEnrollFooter both have exactly same code.. Not sure why it was added and can't be reused.

Reply:
@pradeepdewda-okta - Can you create a ticket so we can clean this up? We're putting these findings under the `widget-help-wanted` label for the Client SDK team?

Reply:
I don't think SDK team is the right team that will do all these cleanups, I think it will be responsibility of home teams. I will create a jira for this though.

https://oktainc.atlassian.net/browse/OKTA-369626